akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerMTSpec.scala

```diff
@@ -36,52 +36,69 @@ class CircuitBreakerMTSpec extends AkkaSpec {
       awaitCond(failingCall())
     }
 
-    def testCallsWithBreaker(breaker: CircuitBreaker): immutable.IndexedSeq[Future[String]] = {
-      val aFewActive = new TestLatch(5)
-      for (_ <- 1 to numberOfTestCalls)
-        yield breaker
-          .withCircuitBreaker(Future {
-            aFewActive.countDown()
-            Await.ready(aFewActive, 5.seconds.dilated)
-            "succeed"
-          })
-          .recoverWith {
-            case _: CircuitBreakerOpenException =>
-              aFewActive.countDown()
-              Future.successful("CBO")
-          }
+    def testCallsWithBreaker(breaker: CircuitBreaker, numberOfCalls: Int): immutable.IndexedSeq[Future[String]] = {
+      for (_ <- 1 to numberOfCalls)
+        yield makeCallWithBreaker(breaker)
     }
 
+    def makeCallWithBreaker(breaker: CircuitBreaker): Future[String] =
+      breaker.withCircuitBreaker(Future.successful("succeed")).recoverWith {
+        case _: CircuitBreakerOpenException =>
+          Future.successful("CBO")
+      }
+
     "allow many calls while in closed state with no errors" in {
-      val futures = testCallsWithBreaker(newBreaker)
+      val futures = testCallsWithBreaker(newBreaker, 10)
       val result = Await.result(Future.sequence(futures), 5.second.dilated)
-      result.size should ===(numberOfTestCalls)
+      result.size should ===(10)
       result.toSet should ===(Set("succeed"))
     }
 
     "transition to open state upon reaching failure limit and fail-fast" in {
       val breaker = newBreaker
       openBreaker(breaker)
-      val futures = testCallsWithBreaker(breaker)
-      val result = Await.result(Future.sequence(futures), 5.second.dilated)
-      result.size should ===(numberOfTestCalls)
-      result.toSet should ===(Set("CBO"))
+
+      breaker.isOpen shouldBe true
+      val call = makeCallWithBreaker(breaker)
+      val result = Await.result(call, 5.second.dilated)
+      result shouldBe "CBO"
```

Comment:
We could imagine that the goal here is to call many times to show that it keeps open. Although, after a while it will move to half-open and the tests will fail. 

So we kind of depend too much on timing.